frontend/js/src/playlists/CreateOrEditPlaylistModal.tsx

```diff
@@ -113,27 +123,53 @@ export default class CreateOrEditPlaylistModal extends React.Component<
     });
   };
 
-  addCollaborator = (evt: React.MouseEvent<HTMLButtonElement>): void => {
-    evt.preventDefault();
+  addCollaborator = (collaborator: string): void => {
     const { collaborators, newCollaborator } = this.state;
-    if (collaborators.indexOf(newCollaborator) !== -1) {
+    if (collaborators.indexOf(collaborator) !== -1) {
       // already in the list
       this.setState({ newCollaborator: "" });
       return;
     }
     this.setState({
-      collaborators: [...collaborators, newCollaborator],
+      collaborators: [...collaborators, collaborator],
       newCollaborator: "",
     });
   };
 
+  searchUsers = async () => {
+    const { currentUser } = this.context;
+    const { newCollaborator } = this.state;
+    if (currentUser?.auth_token) {
+      try {
+        const url = new URL("http://localhost:8100/1/playlist/search/users/");
+        url.searchParams.append("search_term", newCollaborator);
+        const response = await fetch(url.toString(), {
+          method: "GET",
+          headers: {
+            Authorization: `Token ${currentUser.auth_token}`,
+            "Content-Type": "application/json;charset=UTF-8",
+          },
+        });
+
+        const parsedResponse = await response.json();
```

Comment:
This could be extracted as a separate API utility in the [APIService file](https://github.com/metabrainz/listenbrainz-server/blob/master/frontend/js/src/utils/APIService.ts).
It has the advantage for example that you can use `this. APIBaseURI` instead of hardcoding a url (looking at you, `http://localhost:8100` on line 144 😛  ), and it also has some error handling helpers.
You can use any of the methods in there as a starting point: https://github.com/metabrainz/listenbrainz-server/blob/4b3fa44f7555c5ba8dc28e7d7cbf5b2a6c5b51c8/frontend/js/src/utils/APIService.ts#L182-L202